src/common/components/paymentMethods/creditCardForm/creditCardForm.component.js

```diff
@@ -131,24 +119,42 @@ class CreditCardController {
   savePayment(){
     this.creditCardPaymentForm.$setSubmitted();
     if(this.creditCardPaymentForm.$valid){
-      let ccpCreditCardNumber = this.paymentMethod && !this.creditCardPayment.cardNumber ? this.paymentMethod['card-number'] : new (this.paymentValidationService.ccp.CardNumber)(this.creditCardPayment.cardNumber).encrypt();
-      let ccpSecurityCode = this.paymentMethod ? null : new (this.paymentValidationService.ccp.CardSecurityCode)(this.creditCardPayment.securityCode).encrypt();
-      this.onPaymentFormStateChange({
-        $event: {
-          state: 'loading',
-          payload: {
-            creditCard: {
-              address: this.useMailingAddress ? undefined : this.creditCardPayment.address,
-              'card-number': ccpCreditCardNumber,
-              'cardholder-name': this.creditCardPayment.cardholderName,
-              'expiry-month': this.creditCardPayment.expiryMonth,
-              'expiry-year': this.creditCardPayment.expiryYear,
-              ccv: ccpSecurityCode
-            },
-            paymentMethodNumber: this.creditCardPayment.cardNumber ? this.creditCardPayment.cardNumber.slice(-4) : false
-          }
-        }
-      });
+      this.tsysService.getManifest()
+        .mergeMap(data => {
+          cruPayments.creditCard.init(this.envService.get(), data.deviceId || '88812128320102', data.manifest); //TODO: remove hard coded deviceId when added to the manifest api endpoint
+          return this.paymentMethod && !this.creditCardPayment.cardNumber ?
+            Observable.of({ tsepToken: this.paymentMethod['card-number'], maskedCardNumber: this.paymentMethod['card-number'] }) : // Send masked card number when card number is not updated
+            cruPayments.creditCard.encrypt(this.creditCardPayment.cardNumber, this.creditCardPayment.securityCode, this.creditCardPayment.expiryMonth, this.creditCardPayment.expiryYear);
```

Comment:
@wrandall22 I hadn't worked on that in this PR. @mattdrees thought it would be fine to require it on the front end and just ignore it in EP:

> Currently, we can't use a cvv anywhere except checkout with a brand new card. (And actually if the gift is recurring or if there's a comment-to-dsg, the cvv will be ignored because the CC charge will be performed by siebel (later on) instead of EP (right now). But I don't think you need to stop sending the cvv in this case; we'll just ignore it.) There is a decent chance that eventually we'll do a zero-dollar card verification call when the user adds a card via self-service, and that can use the cvv.

We can discuss this more.

If we want to ask for the CVV less, an easy solution for me would be to simplify the rules you wrote to hide the CVV field only when editing a credit card. There would still be the cases where we add a new card and ask for a CVV but don't use it. I could add logic to check if we are in self service when adding a card or if there are only recurring donations or only donations with comments but that logic could get complex, especially if I need to get some of that data from another api request.

Currently I'm collecting the CVV most places but omitting the CVV on any API requests except for the enhancedPurchaseForm. I'm also collecting the CVV when the user enters a new number while editing but hiding the CVV field until they've typed a valid number.

What direction do you guys think we should go? Everything should be doable for me but figuring out if a cart has gifts that qualify for realtime cc transactions could be more difficult, especially since the user can go back and add/remove anything from their cart.